programs/net-util/net-util.py

```diff
@@ -0,0 +1,371 @@
+                                bandwidths[host] = stats
+                            elif fieldName == 'bytes_sent':
+                                bytesSent = int(sample.value)
+                                stats = bandwidths.get(host, bandwidthStats())
+                                stats.bytesSent = bytesSent
+                                bandwidths[host] = stats
+                            elif fieldName == 'connection_start_time':
+                                connectionStarted = int(sample.value)
+                                stats = bandwidths.get(host, bandwidthStats())
+                                stats.connectionStarted = connectionStarted
+                                bandwidths[host] = stats
+                            else:
+                                attrname = fieldName[:1] + fieldName.replace('_', ' ').title().replace(' ', '')[1:] + 'LW'
+                                if hasattr(self, attrname):
+                                    listwalker = getattr(self, attrname)
+                                    listwalker[startOffset:endOffset] = [urwid.Text(self.peerMetricConversions[fieldName](sample.value))]
+                                else:
+                                    listwalker = getattr(self, 'hostnameLW')
+                                    listwalker[startOffset:endOffset] = [urwid.Text(fieldName.replace('_', '.'))]
+                    elif sample.name == 'nodeos_info':
+                        fieldName = self.fields.get(self.prometheusMetrics[(sample.name, 'server_version_string')])
+                        field = getattr(self, fieldName)
+                        field.set_label(sample.labels['server_version_string'])
+                    else:
+                        if sample.name not in self.ignoredPrometheusMetrics:
+                            logger.warning(f'Received unhandled Prometheus metric {sample.name}')
+                else:
+                    if sample.name == 'nodeos_p2p_connections':
+                        now = time.time_ns()
+                        hostListwalker = getattr(self, 'ipAddressLW')
+                        for host, stats in bandwidths.items():
+                            startOffset = hostListwalker.index(host)
+                            endOffset = startOffset + 1
+                            connected_seconds = (now - stats.connectionStarted)/1000000000
+                            listwalker = getattr(self, 'receiveBandwidthLW')
+                            bps = stats.bytesReceived/connected_seconds
+                            listwalker[startOffset:endOffset] = [urwid.Text(humanReadableBytesPerSecond(bps))]
+                            listwalker = getattr(self, 'sendBandwidthLW')
+                            bps = stats.bytesSent/connected_seconds
+                            listwalker[startOffset:endOffset] = [urwid.Text(humanReadableBytesPerSecond(bps))]
+        mainLoop.set_alarm_in(0.5, self.update)
```

Comment:
This queries nodeos and re-displays every half second, but most of the data is only being updated every 30 seconds.

Reply:
Added `--refresh-interval` command line option.